### ISSUE TITLE
Add support for multi-CMS security guards (Joomla + WordPress)

### ISSUE THREAD

**Issue by fsjn** [joomla-extensions/jedchecker#269]
### Description

The JEXEC rule (`PH2`) does not recognize valid security guards used in extensions that support multiple CMS platforms (Joomla + WordPress).

### Example

Extensions built for both Joomla and WordPress (e.g., YOOtheme Pro elements) use this pattern:
```php
<?php
defined('_JEXEC') or defined('ABSPATH') or die();
```

This is a valid and secure guard that:
1. Checks `_JEXEC` (Joomla)
2. Checks `ABSPATH` (WordPress)
3. Calls `die()` if neither is defined

### Expected Behavior

JED Checker should recognize this as a valid security guard.

### Actual Behavior

JED Checker reports: "The JEXEC security check was not found in this file."

### Root Cause

The regex in `jexec.php` expects `die`/`exit` immediately after the first `defined()` check:
```php
. '\\\\?defined ?\( ?'
. '([\'"])(?:' . implode('|', $defines) . ')\1'
. ' ?\) ?(?:or |\|\| ?)(?:die|exit)\b'
```

This pattern matches:
```php
defined('_JEXEC') or die();  ✅
```

But not:
```php
defined('_JEXEC') or defined('ABSPATH') or die();  ❌
```

### Suggested Fix

Update the regex to allow additional `defined()` checks between the primary constant and `die`/`exit`:
```php
$additionalDefined = '(?: ?(?:or |\|\| ?)\\\\?defined ?\( ?[\'"][A-Za-z_][A-Za-z0-9_]*[\'"] ?\))*';

$this->regex
    = '#^\s*'
    . '<\?php\s+'
    . '(?:declare ?\(strict_types ?= ?1 ?\) ?; ?)?'
    . '(?:namespace [0-9A-Za-z_\\\\]+ ?; ?)?'
    . '(?:use (?:function |const )?' . $use_ns . '; ?)*'
    . '\\\\?defined ?\( ?'
    . '([\'"])(?:' . implode('|', $defines) . ')\1'
    . ' ?\)'
    . $additionalDefined  // Support chained defined() checks
    . ' ?(?:or |\|\| ?)(?:die|exit)\b'
    . '#i';
```

This would support:
- `defined('_JEXEC') or die();` ✅
- `defined('_JEXEC') || die();` ✅
- `defined('_JEXEC') or defined('ABSPATH') or die();` ✅
- `defined('_JEXEC') || defined('ABSPATH') || die();` ✅

### Use Case

YOOtheme Pro is a popular page builder that works on both Joomla and WordPress. Extensions developed for YOOtheme Pro need to support both platforms with a single codebase, requiring multi-CMS security guards.

### Environment

- JED Checker version: 2.4.x
- PHP version: 8.1+
- Joomla version: 5.x
